staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -36,6 +37,15 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
+// StartUntil starts f in a new goroutine and calls done once f has finished.
+func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
```

Comment:
This is an odd function, with an odd name. It looks like the "f" parameter could instead be of type `func()`, and callers could close over a "stop" channel if so desired.

The documentation mentions "call[ing] done," but what is "done" here? Do you mean `sync.(*WaitGroup).Done`?

Consider an alternative:
```go
func RunWithinGroup(f func(), wg *sync.WaitGroup) {
  wg.Add(1)
  go func() {
    defer wg.Done()
    f()
  }()
}
```